configargparse.py

```diff
@@ -813,19 +927,68 @@ def parse_known_args(
             env_vars=os.environ,
             ignore_help_args=False,
     ):
-        """Supports all the same args as the `argparse.ArgumentParser.parse_args()`,
+        """Supports all the same args as the `argparse.ArgumentParser.parse_known_args()`,
         as well as the following additional args.
 
         Arguments:
             args: a list of args as in argparse, or a string (eg. "-x -y bla")
             config_file_contents (str). Used for testing.
             env_vars (dict). Used for testing.
-            ignore_help_args (bool): This flag determines behavior when user specifies ``--help`` or ``-h``. If False,
-                it will have the default behavior - printing help and exiting. If True, it won't do either.
+            ignore_help_args (bool): This flag determines behavior when user specifies ``--help``
+                or ``-h``. If False, it will have the default behavior - printing help and exiting.
+                If True, it won't do either.
 
         Returns:
             tuple[argparse.Namespace, list[str]]: tuple namescpace, unknown_args
         """
+        return self._parse_known_args_cap(intermixed=False,
+                                          args=args,
+                                          namespace=namespace,
+                                          config_file_contents=config_file_contents,
+                                          env_vars=env_vars,
+                                          ignore_help_args=ignore_help_args)
+
+    def parse_known_intermixed_args(
+            self,
+            args=None,
+            namespace=None,
+            config_file_contents=None,
+            env_vars=os.environ,
+            ignore_help_args=False,
+    ):
+        """Supports all the same args as the `argparse.ArgumentParser.parse_known_intermixed_args()`,
+        as well as the following additional args.
+
+        Arguments:
+            args: a list of args as in argparse, or a string (eg. "-x blah -y blaz")
+            config_file_contents (str). Used for testing.
+            env_vars (dict). Used for testing.
+            ignore_help_args (bool): This flag determines behavior when user specifies ``--help``
+                or ``-h``. If False, it will have the default behavior - printing help and exiting.
+                If True, it won't do either.
+
+        Returns:
+            tuple[argparse.Namespace, list[str]]: tuple namescpace, unknown_args
+        """
+        return self._parse_known_args_cap(intermixed=True,
+                                          args=args,
+                                          namespace=namespace,
+                                          config_file_contents=config_file_contents,
+                                          env_vars=env_vars,
+                                          ignore_help_args=ignore_help_args)
+
+    def _parse_known_args_cap(
```

Comment:
Not sure what '_cap' suffix stands for - maybe _parse_known_args_impl is better name?

Reply:
cap = ConfigArgParse since this function wraps the corresponding function from the base class, rather than overriding it. I agree that if not renamed this should at least be explicit in the docstring.

Reply:
would prefer replacing `_cap` with `_impl` or `_base` or something